core/src/main/java/io/questdb/griffin/engine/groupby/GroupByNotKeyedRecordCursorFactory.java

```diff
@@ -162,6 +175,13 @@ public boolean hasNext() {
                 while (baseCursor.hasNext()) {
                     circuitBreaker.statefulThrowExceptionIfTripped();
                     groupByFunctionsUpdater.updateExisting(simpleMapValue, baseRecord);
+                    boolean earlyExit = true;
```

Comment:
is it possible to override a single `hasNext` instead of a full copy of the `GroupByNotKeyedRecordCursor`?

Reply:
Good idea. Done in 16762a9083fe85ede985ef35eb35c0d81cf92b68